contrib/lite-setup.py

```diff
@@ -0,0 +1,142 @@
+ldap_uri = ldap://{args.servername}
+mode = production
+debug = False
+enable_ra = True
+ra_plugin = dogtag
+dogtag_version = 10
+"""
+
+KRB5_CONF = """\
+[libdefaults]
+ default_realm = {args.realm}
+ dns_lookup_realm = false
+ dns_lookup_kdc = false
+ rdns = false
+ ticket_lifetime = 24h
+ forwardable = true
+ udp_preference_limit = 0
+ default_ccache_name = FILE:{args.ccache}
+
+[realms]
+ {args.realm} = {{
+  kdc = {args.servername}:88
+  master_kdc = {args.servername}:88
+  admin_server = {args.servername}:749
+  # kdc = https://{args.servername}/KdcProxy
+  # master_kdc = https://{args.servername}/KdcProxy
+  # admin_server = https://{args.servername}/KdcProxy
+  default_domain = ipa.example
+  pkinit_anchors = FILE:{args.ca_crt}
+  pkinit_pool = FILE:{args.ca_crt}
+  http_anchors = FILE:{args.ca_crt}
+}}
+
+[domain_realm]
+ .ipa.example = {args.realm}
+ ipa.example = {args.realm}
+ {args.servername} = {args.realm}
+"""
+
+ACTIVATE = """\
+deactivate_ipaenv () {{
+
+    export PS1="${{_OLD_PS1}}"
+    unset _OLD_PS1
+    unset KRB5_CONFIG
+    unset KRB5CCNAME
+    unset IPA_CONFDIR
+    unset PYTHONPATH
+    unset -f deactivate_ipaenv
+}}
+
+export _OLD_PS1="${{PS1:-}}"
+export PS1="(ipaenv) ${{PS1:-}}"
+export KRB5_CONFIG="{args.krb5_conf}"
+export KRB5CCNAME="{args.ccache}"
+export IPA_CONFDIR="{args.dot_ipa}"
+export PYTHONPATH="{args.basedir}"
+"""
+
+MSG = """\
+Configured for server '{args.servername}' and realm '{args.realm}'.
+
+To activate the IPA test env:
+
+    source {args.activate}
+    kinit
+    make lite-server
+    python3 -m ipaclient ping
+
+To deactivate the IPA test env and to unset the env vars:
+
+    deactivate_ipaenv
+
+The source file configures the env vars:
+
+    export KRB5_CONFIG="{args.krb5_conf}"
+    export KRB5CCNAME="{args.ccache}"
+    export IPA_CONFDIR="{args.dot_ipa}"
+    export PYTHONPATH="{args.basedir}"
+"""
+
+parser = argparse.ArgumentParser()
+parser.add_argument("servername")
+parser.add_argument("domain", default=None, nargs="?")
+
+
+def main():
+    args = parser.parse_args()
+    if args.domain is None:
+        args.domain = args.servername.split(".", 1)[1]
+    args.hostname = socket.gethostname()
+    args.domain = args.domain.lower().rstrip(".")
+    args.realm = args.domain.upper()
+    args.basedn = ",".join(f"dc={part}" for part in args.domain.split("."))
+
+    args.basedir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    args.dot_ipa = os.path.expanduser("~/.ipa")
+    args.default_conf = os.path.join(args.dot_ipa, "default.conf")
+    args.ca_crt = os.path.join(args.dot_ipa, "ca.crt")
+    args.krb5_conf = os.path.join(args.dot_ipa, "krb5.conf")
+    args.ccache = os.path.join(args.dot_ipa, "ccache")
+    args.activate = os.path.join(args.dot_ipa, "activate.sh")
+
+    if not os.path.isdir(args.dot_ipa):
+        os.makedirs(args.dot_ipa, mode=0o750)
+
+    with urlopen(f"http://{args.servername}/ipa/config/ca.crt") as req:
+        ca_data = req.read()
```

Comment:
Why not just copy from /etc/ipa/ca.crt? It has to run locally because localhost is specified as the xmlrpc_url so we know that the file is there.

Reply:
I have split up the PR and moved the new lite-setup to a different PR.

The script does not assume local configuration. The ``xmlrpc_uri`` is ``http://localhost:8888``, which is the hostname and port of a local lite-server setup.